src/Compilers/CSharp/Portable/Symbols/Source/SourceExtensionTypeSymbol.cs

```diff
@@ -267,74 +197,6 @@ internal sealed override ImmutableArray<NamedTypeSymbol> GetDeclaredBaseExtensio
-                {
-                    var diagnostics = BindingDiagnosticBag.GetInstance();
-                    var acyclicBaseExtensions = makeAcyclicBaseExtensions(diagnostics);
-
-                    if (ImmutableInterlocked.InterlockedInitialize(ref _lazyBaseExtensions, acyclicBaseExtensions))
-                    {
-                        AddDeclarationDiagnostics(diagnostics);
-                    }
-                    diagnostics.Free();
-                }
-
-                return _lazyBaseExtensions;
-
-                ImmutableArray<NamedTypeSymbol> makeAcyclicBaseExtensions(BindingDiagnosticBag diagnostics)
-                {
-                    ImmutableArray<NamedTypeSymbol> declaredBaseExtensions = GetDeclaredExtensionInfo(basesBeingResolved: null).BaseExtensions;
-
-                    var result = ArrayBuilder<NamedTypeSymbol>.GetInstance();
-                    foreach (var declaredBaseExtension in declaredBaseExtensions)
-                    {
-                        if (BaseTypeAnalysis.TypeDependsOn(depends: declaredBaseExtension, on: this))
-                        {
-                            result.Add(new ExtendedErrorTypeSymbol(declaredBaseExtension, LookupResultKind.NotReferencable,
-                                diagnostics.Add(ErrorCode.ERR_CycleInBaseExtensions, Locations[0], this, declaredBaseExtension)));
-                            continue;
-                        }
-
-                        result.Add(declaredBaseExtension);
-
-                        if (declaredBaseExtension.ContainingModule != this.ContainingModule)
-                        {
-                            var useSiteInfo = new CompoundUseSiteInfo<AssemblySymbol>(diagnostics, ContainingAssembly);
-                            declaredBaseExtension.AddUseSiteInfo(ref useSiteInfo);
-
-                            foreach (var extension in declaredBaseExtension.AllBaseExtensionsNoUseSiteDiagnostics)
-                            {
-                                if (extension.ContainingModule != this.ContainingModule)
-                                {
-                                    extension.AddUseSiteInfo(ref useSiteInfo);
-                                }
-                            }
-                            diagnostics.Add(Locations[0], useSiteInfo);
-                        }
-                    }
-
-                    return result.ToImmutableAndFree();
-                }
-            }
-        }
-
-        internal override ImmutableArray<NamedTypeSymbol> AllBaseExtensionsNoUseSiteDiagnostics
-        {
-            get
-            {
-                if (_lazyAllBaseExtensions.IsDefault)
-                {
-                    ImmutableInterlocked.InterlockedInitialize(ref _lazyAllBaseExtensions, MakeAllBaseExtensions());
-                }
-
-                return _lazyAllBaseExtensions;
-            }
-        }
-
         private ExtensionInfo GetDeclaredExtensionInfo(ConsList<TypeSymbol>? basesBeingResolved)
```

Comment:
Looks like `basesBeingResolved` is always `null` now. Consider adding a PROTOTYPE comment.